pyfar/dsp/dsp.py

```diff
@@ -1961,6 +1961,15 @@ def average(signal, mode='linear', caxis=None, weights=None, keepdims=False):
         raise ValueError((
             f"mode is '{mode}' and signal is type '{signal.__class__}'"
             " but must be of type 'Signal' or 'FrequencyData'."))
+    if ((type(signal) == pyfar.TimeData or type(signal) == pyfar.Signal)
+        and signal.complex and mode in (
+                                        'log_magnitude_zerophase',
+                                        'magnitude_zerophase',
+                                        'magnitude_phase',
+                                        'power',)):
+
+        raise ValueError((
+            f"mode '{mode}' is not defined for complex signals."))
```

Comment:
what does not defined mean? according to the description above it should be defined, or does it mean it's not implemented?

Reply:
If I remember our discussion correctly, we did not check in depth if there is a usefull definition and leave it up to people who need that to implement in case there is one.